server/server_test.go

```diff
@@ -146,22 +146,20 @@ func TestIdleClientConnections(t *testing.T) {
 
 	okFn := func(conn net.Conn, originURL *url.URL) {
 		time.Sleep(time.Millisecond * 900)
-		conn.Write([]byte("GET / HTTP/1.1\r\nHost: www.google.com\r\n\r\n"))
+		conn.Write([]byte("GET / HTTP/1.1\r\nHost: " + originURL.Host + "\r\n\r\n"))
```

Comment:
I'm not sure if it worked before, but the test is supposed to hit the test origin server. Without this change, `conn.Read` below always gets `EOF`.